### PR TITLE
refactor(http): remove dead code and improve `UserAgent` testing

### PR DESCRIPTION
**Pull request by kt3k**
This PR removes some dead code and improves typings in `http/user_agent.ts`, and also adds test cases which exercise uncovered lines.

towards #3713

### CODE REVIEW

File: http/user_agent.ts [hunk 1]
```diff
@@ -146,15 +145,7 @@ const windowsVersionMap = new Map<string, string | string[]>([
...
 
-function has(str1: string | string[], str2: string): boolean {
-  if (Array.isArray(str1)) {
-    for (const el of str1) {
-      if (lowerize(el) === lowerize(str2)) {
-        return true;
-      }
-    }
-    return false;
-  }
```
Comment (kt3k) [inline http/user_agent.ts:157]: `str1` is always string with all `has` usages. This if-block is dead code.

File: http/user_agent.ts [hunk 2]
```diff
@@ -208,9 +196,6 @@ function mapper(
               target[prop] = match ? match.replace(re, value) : undefined;
             } else {
               const [prop, re, value, fn] = processor;
-              if (!fn) {
-                throw new TypeError("Function must be defined in processor");
-              }
```
Comment (kt3k) [inline http/user_agent.ts:213]: If `processor` has 4 elements, the 4th element needs to be a function by its typing. So this if-block does nothing.

File: http/user_agent.ts [hunk 3]
```diff
@@ -146,15 +145,7 @@ const windowsVersionMap = new Map<string, string | string[]>([
...
-  if (Array.isArray(str1)) {
-    for (const el of str1) {
-      if (lowerize(el) === lowerize(str2)) {
-        return true;
-      }
-    }
-    return false;
-  }
+function has(str1: string, str2: string): boolean {
   return lowerize(str2).indexOf(lowerize(str1)) !== -1;
```
Comment (iuioiua) [inline http/user_agent.ts:149]: How about `.includes()`?

File: http/user_agent.ts [hunk 4]
```diff
@@ -184,21 +175,18 @@ function mapper(
...
     let k = 0;
     while (j < matchers.length && !matches) {
-      if (!matchers[j]) {
-        break;
-      }
       matches = matchers[j++]!.exec(ua);
 
       if (matches) {
         for (const processor of processors) {
-          const match = matches[++k]!;
```
Comment (kt3k) [inline http/user_agent.ts:194]: There were cases where `match` is actually `undefined`. So appending `!` here is not ideal.

File: http/user_agent.ts
Comment (iuioiua) [inline http/user_agent.ts:1]: Can we remove `lowerize()` and `trim()`? Are there any others we can remove?

**Review by iuioiua — CHANGES_REQUESTED** [review]
Few nits

File: http/user_agent.ts
Comment (kt3k) [reply http/user_agent.ts:1]: lowerize and trim is used as `ProcessingFn` in processors. I think it's difficult to remove them

**Review by iuioiua — APPROVED** [review]

**Pull request merged**